wi-fi/hal/cyabs_rtos.c

```diff
@@ -196,7 +215,14 @@ cy_rslt_t cy_rtos_set_semaphore(cy_semaphore_t *semaphore, bool in_isr)
 	if (semaphore == NULL)
 		return CY_RTOS_BAD_PARAM;
 
-	mutexUnlock(*semaphore);
+	mutexLock(semaphore->mutex);
+
+	if (semaphore->v < semaphore->m) {
+		condSignal(semaphore->cond);
+		++semaphore->v;
```

Comment:
Doesn't really matter, as we still have this lock, but signaling after changing v value would better explain intention imho